lib/Map/AddressConverter.js

```diff
@@ -0,0 +1,123 @@
+     * @type {CorsProxy}
+     */
+    this.corsProxy = corsProxy;
+};
+
+/**
+ * Convert addresses one at a time, making multiple calls to the server. For some geocoding services, this is all they
+ * provide. When the promise fulfills, the tableStructure has been updated with new lat and lon columns.
+ *
+ * @return {Promise} Promise with no return value.
+ */
+AddressConverter.prototype.convertEachAddress = function() {
+
+    if (!this.tableStructure.hasAddress) {
+        throw new DeveloperError('This tableStructure has no addresses!');
+    }
+
+    var addressesCol = this.tableStructure.columnsByType[VarType.ADDR][0];
+    if (addressesCol.length === 0) {
+        throw new DeveloperError('Even though the tableStructure reports it has an address column, '
+                                 + 'it has no addresses!');
+    }
+    var suburbs = this.tableStructure.getColumnWithName("Suburb");
+    smooshColumn(addressesCol, suburbs);
+    var state = this.tableStructure.getColumnWithName("State");
+    smooshColumn(addressesCol, state);
+    var postCodeCol = this.tableStructure.getColumnWithName("Postcode");
+    smooshColumn(addressesCol, postCodeCol);
+
+    var that = this;
+    var promises = addressesCol.values.map(function(address) { return convertAddress(address, that.corsProxy); });
+    return when.all(promises).then( function(results) {
+        var longValues = results.map(function(latLong) { return latLong.longitude; });
+        that.tableStructure.addColumn("Lon", longValues);
+
+        var latValues = results.map(function(latLong) { return latLong.latitude; });
+        that.tableStructure.addColumn("Lat", latValues);
+        return when(true);
+    }).otherwise(function(e) {
+        console.log("Unable to create lat and lon columns for TableStructure.");
+        console.log(e);
+    });
+};
+
+/**
+ * Add info to address, comma separated.
+ * @param {TableColumn} [addressesCol] Table column with addresses to be added to
+ * @param {TableColumn} [newCol] Table column with extra info to be added to addresses column
+ */
+function smooshColumn(addressesCol, newCol) {
+    if (!defined(newCol)) {
+        return;
+    }
+    if (newCol.values.length === addressesCol.values.length) {
+        for (var i=0; i<addressesCol.values.length; i++) {
+            if (addressesCol.values[i] !== null && newCol.values[i] !== null) {
+                addressesCol.values[i] = addressesCol.values[i] + "," + newCol.values[i];
+            }
+        }
+    }
+}
+
+/**
+ * A promise to convert a single address to lat long coordinates, which resolves to
+ * a length two array with (longitude, latitude) coordinates.
+ *
+ * @param {String} [address] The address to convert
+ * @param {String} [corsProxy] Proxy for cross origin resource sharing
+ * @return {Promise} A promise that eventually returns a pair of coordinates.
+ */
+function convertAddress(address, corsProxy) {
+
+    var gnafApi = new GNAFApi(corsProxy);
```

Comment:
This is very specific to the Australian GNAF.  Either the name (`AddressConverter`) should be changed to reflect that, or it should be made more generic.